snapcraft/internal/parser.py

```diff
@@ -216,16 +220,25 @@ def _process_index(output):
 
     output = output.replace(b"{{{", b"").replace(b"}}}", b"")
 
-    all_data = yaml.load_all(output)
-    for data in all_data:
-        try:
-            parts_list, after_parts = _process_entry(data)
-        except InvalidEntryError as e:
-            logger.warning('Invalid wiki entry: {}'.format(e))
-            continue
-
-        if is_valid_parts_list(parts_list, after_parts):
-            master_parts_list.update(parts_list)
+    # split the wiki into an array of entries to allow the parser to
+    # proceed when invalid yaml is found.
+    entries = output.split(b'\n---\n')
```

Comment:
I've added a bug https://bugs.launchpad.net/snapcraft/+bug/1599974 to track that change.